src/main/java/org/fogbowcloud/saps/engine/utils/retry/catalog/CatalogRetry.java

```diff
@@ -4,5 +4,5 @@
 
 public interface CatalogRetry<T> {
 
-	public T run() throws SQLException;
+	T run() throws SQLException;
```

Comment:
it seems odd to throw an SQLException in this class? how is this related to SQL

Reply:
I created an exception class (CatalogRetryException) to represent errors coming from that interface